File: app/src/main/java/com/tdev/mplayr/data/Song.java
package com.tdev.mplayr.data;

public class Song {
    public final long id;
    public final String title;
    public final String artist;
    public final String album;
    public final long duration;
    public final String path;

    public Song(long id, String title, String artist, String album, long duration, String path) {
        this.id = id;
        this.title = title != null && !title.isEmpty() ? title : "Unknown";
        this.artist = artist != null && !artist.isEmpty() ? artist : "Unknown";
        this.album = album != null && !album.isEmpty() ? album : "Unknown";
        this.duration = duration;
        this.path = path;
    }

    public String formatDuration() {
        long secs = duration / 1000;
        return String.format("%d:%02d", secs / 60, secs % 60);
    }
}

File: app/src/main/kotlin/com/tdev/mplayr/data/Song.kt
package com.tdev.mplayr.data

import android.content.ContentUris
import android.net.Uri

data class Song(
    val id: Long,
    val title: String,
    val artist: String,
    val album: String,
    val albumId: Long,
    val duration: Long,
    val uri: Uri,          // content:// Uri — DATA kolonu yerine
    val filePath: String = "", // [27][19] klasör kara listesi ve kopya tespiti için dosya yolu
    val genre: String = "Bilinmiyor" // [16] Tür dağılımı pasta grafiği için
) {
    val artUri: Uri get() = ContentUris.withAppendedId(
        Uri.parse("content://media/external/audio/albumart"), albumId
    )

    fun formatDuration(): String {
        val secs = duration / 1000
        return "%d:%02d".format(secs / 60, secs % 60)
    }
}
